src/app/beer_garden/systems.py

```diff
@@ -263,31 +268,50 @@ def purge_system(
     if force and not system.local:
         return remove_system(system=system)
 
-    # Publish stop message to all instances of this system
-    publish_stop(system)
-
-    # If lpm is managing the runner, wait for the runners to stop
-    for inst in system.instances:
-        if lpm.has_instance_id(inst.id):
-            lpm.remove(instance_id=inst.id)
-
-    system = db.reload(system)
-
-    # Now clean up the message queues. It's possible for the request or admin queue to
-    # be none if we are stopping an instance that was not properly started.
     for instance in system.instances:
-        force_disconnect = instance.status != "STOPPED"
+        if instance.status not in ("STOPPED", "DEAD"):
+            if lpm.has_instance_id(instance.id):
+                try:
+                    lpm.remove(instance_id=instance.id)
+                except Exception as ex:
+                    if not force:
+                        raise PluginError(
+                            f"Error attempting to stop {system}[{instance.name}]"
+                        ) from ex
+
+                    logger.warning(
+                        f"Error while stopping instance {system}[{instance.name}]. "
+                        f"Force flag was specified so system delete will continue. "
+                        f"Underlying exception was: {ex}"
+                    )
+            else:
+                try:
+                    # TODO - It would be nice to wait for the instance to stop
+                    publish_stop(system, instance=instance)
+                except Exception as ex:
+                    if not force:
+                        raise RequestPublishException(
+                            f"Error attempting to publish stop message for "
+                            f"{system}[{instance.name}"
+                        ) from ex
+
+                    logger.warning(
+                        f"Error while publishing stop message for "
+                        f"{system}[{instance.name}]. Force flag was specified so "
+                        f"system delete will continue. Underlying exception was: {ex}"
+                    )
 
         request_queue = ""
         try:
             request_queue = instance.queue_info.get("request", {}).get("name")
             if request_queue:
-                queue.remove(
-                    request_queue, force_disconnect=force_disconnect, clear_queue=True
-                )
+                queue.remove(request_queue, force_disconnect=force, clear_queue=True)
         except Exception as ex:
             if not force:
-                raise
+                raise PluginError(
+                    f"Error attempting to remove request queue '{request_queue}' for "
+                    f"{system}[{instance.name}]"
+                ) from ex
```

Comment:
I did not know this was an option, I will have to keep that in mind next time I raise custom exceptions.